### DISCUSSION TITLE
still unsure how to get balance from Digital Asset

### DISCUSSION THREAD

**Discussion by dellwatson** [aptos-labs/aptos-developer-discussions#55] [Questions] [answered]
I only get how to get the address instead of balance of my total NFT

https://github.com/aptos-labs/aptos-core/blob/93ed9aefff550250ecc620651cd3eb4cf37ecc40/aptos-move/framework/aptos-token-objects/sources/token.move#L280C1-L283C6

```js
    /// Generates the token's address based upon the creator's address, the collection's name and the token's name.
    public fun create_token_address(creator: &address, collection: &String, name: &String): address {
        object::create_object_address(creator, create_token_seed(collection, name))
    }
    ```
    
    I dont find function that return the number of balance here.
    why not make it same like from EIP-712 ? with token index, ownerOf etc

**Comment by 0x-j** [comment, marked as answer]
The only way to get your NFT balance in a collection is to query indexer, because Aptos's digital asset standard (and fungible asset) is based on [object](https://aptos.dev/standards/aptos-object), i.e. it's more like UTXO model where we try to store data under each user instead of like account model where we store all users' data in one big table. 
```typescript
  import { Aptos, AptosConfig, Network } from "@aptos-labs/ts-sdk";
  const config = new AptosConfig({
    network: Network.TESTNET,
  });
  export const aptos = new Aptos(config);

  const result = await aptos.getAccountOwnedTokensFromCollectionAddress({
    accountAddress: ownerAddr,
    collectionAddress: COLLECTION_ID,
  });

  console.log("my nfts in the collection", result);
```

**Reply by dellwatson** [reply]
so it's unable to get balance on-chain? since i need the smart contract to detect balance of the collections

**Reply by 0x-j** [reply]
If you want the total supply of NFT in a collection, you can do this
```rust
use aptos_token_objects::collection::Collection;
use aptos_framework::object::address_to_object;

let collection_object = address_to_object<Collection>(collection_address);
collection::count(collection_object)
```
If you want to count how many NFT an address has, unfortunately you can not get that on chain, you have to use indexer
